packages/cozy-konnector-libs/src/libs/saveFiles.spec.js

```diff
@@ -267,3 +273,62 @@ describe('saveFiles', function() {
     })
   })
 })
+
+describe('getFileIfExists', function() {
+  describe('when in filepath mode', () => {
+    beforeEach(function() {
+      manifest.data.slug = false
+      cozyClient.files.statByPath.mockReset()
+    })
+    it('when the file does not exist, should not return any file', async () => {
+      cozyClient.files.statByPath.mockReturnValue(asyncReject(false))
+      const result = await getFileIfExists(
+        { filename: 'testfile.txt' },
+        { folderPath: '/test/path' }
+      )
+      expect(result).toBe(false)
+    })
+    it('when the file exists and without metadata, should return the file', async () => {
+      cozyClient.files.statByPath.mockReturnValue(
+        asyncResolve({ name: 'testfile.txt' })
+      )
+      const result = await getFileIfExists(
+        { filename: 'testfile.txt' },
+        { folderPath: '/test/path' }
+      )
+      expect(result).toEqual({ name: 'testfile.txt' })
+    })
+  })
+  describe('when in metadata mode', () => {
+    const options = {
+      filePrimaryKeys: ['vendorRef'],
+      sourceAccountOptions: {
+        sourceAccountIdentifier: 'accountidentifier'
+      },
+      folderPath: '/test/path'
+    }
+    beforeEach(function() {
+      manifest.data.slug = 'testconnector'
+      cozyClient.data.defineIndex.mockReturnValue(asyncResolve('index'))
+      cozyClient.files.statByPath.mockReset()
+      queryAll.mockReset()
+    })
+    it('when the file does not exist, should not return any file', async () => {
+      queryAll.mockReturnValue(asyncResolve([]))
+      cozyClient.files.statByPath.mockReturnValue(asyncReject(false))
+      const result = await getFileIfExists(
+        { filename: 'testfile.txt' },
+        options
+      )
+      expect(result).toBe(false)
+    })
+    it('when the file exists, should return the file', async () => {
```

Comment:
We only test the case with one attribute, it could be interesting to add more valuable tests :
ex:  
 - input an array with 2 attributes for deduplication
 - Check that attributes is correctly processed (trough the 'calculate', and then when passed to 'queryAll')